presto-main/src/main/java/com/facebook/presto/server/ServerMainModule.java

```diff
@@ -328,6 +331,22 @@ protected void setup(Binder binder)
 
         // dynamic filter summary
         jsonCodecBinder(binder).bindJsonCodec(DynamicFilterSummary.class);
+        binder.bind(DynamicFilterService.class).in(Scopes.SINGLETON);
+        jaxrsBinder(binder).bind(DynamicFilterResource.class);
+
+        if (featuresConfig.isDynamicPartitionPruningEnabled()) {
+            binder.bind(new TypeLiteral<Optional<DynamicFilterClientSupplier>>() {}).toProvider(DynamicFilterClientFactoryProvider.class).in(Scopes.SINGLETON);
```

Comment:
`Optional` was to simplify tests not targeting DPP. E.g. `TaskTestUtils`.
I don't feel strong about it, if you don't like it.